src/event/stream.rs

```diff
@@ -113,7 +113,9 @@ impl Stream for EventStream {
             Ok(false) => {
                 if !self
```

Comment:
I am not entirely sure about the semantics of this code snippet. `compare_and_swap` (per docs): `The return value is always the previous value. If it is equal to current, then the value was updated`. 

So this checks that the previous value was false, am I right? In that case, maybe changing the below code to

```rust
if let Ok(value) = self.stream_wake_task_executed.compare_exchange(false, true, Ordering::SeqCst, Ordering::SeqCst) {
	if !value {
		//...
	}
}
```

Might show the intention of the code a little bit better.

Reply:
Perfect to me.